### PR TITLE
Add namespace provision request

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Added namespace provision request before each workspace/devworkspace related, factory resolve operation.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/20167

### CODE REVIEW

File: packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts [hunk 1]
```diff
@@ -436,11 +440,10 @@ export class DevWorkspaceClient extends WorkspaceClient {
...
   /**
    * Initialize the given namespace
-   * @param namespace The namespace you want to initialize
-   * @returns If the namespace has been initialized
+   * @returns If the namespace has been provisioned
    */
-  async initializeNamespace(namespace: string): Promise<void> {
-    return DwCheApi.initializeNamespace(namespace);
+  async provisionKubernetesNamespace(): Promise<void> {
+    await this.workspaceClient.restApiClient.provisionKubernetesNamespace();
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts:446]: I consider this should be an action in the InfrastructureNamespaces store. WDYT?

**Review by sleshchenko — APPROVED** [review]
Tested and seems to work as expected from dashboard point of view.
Actually, Che Server failed to provision namespace if it's not there, but it's not related to this PR. I'll report Platform team.

File: packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts (hunk 1 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts:446]: I agree. I have fixed it.

**Comment by sleshchenko** [comment]
Happy path test failure does not seem to be related to your PR changes:
Screenshot_20210922_173849

**Pull request merged**